cmd/maya-exporter/app/collector/jivacollector.go

```diff
@@ -183,11 +187,26 @@ func (e *VolumeExporter) Describe(ch chan<- *prometheus.Desc) {
 // concurrent readers.
 
 // Collect collects all the registered stats metrics from the OpenEBS volumes.
+// It tries to reconnect with the volume if there is any error via a goroutine.
 func (e *VolumeExporter) Collect(ch chan<- prometheus.Metric) {
-	if err := e.collect(); err != nil {
-		return
+	// verify if controller url is not empty
+	if len(e.VolumeControllerURL) != 0 {
+		// collect the metrics from jiva controller and send it via channels
+		if err := e.collect(); err != nil {
+			glog.Error("Error in collecting metrics, found error:", err)
+			return
+		}
+	}
+	// verify if net.Conn is not nil and proceed to collect metrics from
+	// socket.
+	if e.Conn != nil {
```

Comment:
What kind of connection is this? Is this socket connection?
can we rename to be more explicit ?

Reply:
Conn is very generic field of VolumeExporter struct, today we are using "unix" sockets, may be tomorrow we will be using "tcp" for some purpose, so having a generic name would be good and comments will be added explaining about this.